pylint/checkers/magic_methods.py

```diff
@@ -0,0 +1,144 @@
+from typing import TYPE_CHECKING
+
+from astroid import nodes
+
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+class ManualMagicMethodChecker(BaseChecker):
+    """Check for manual __magic__ method calls."""
+
+    __implements__ = IAstroidChecker
+
+    # Docs: https://docs.python.org/3/reference/datamodel.html#basic-customization
+    # We exclude __init__, __new__, __subclasses__, __init_subclass__,
+    # __set_name__, __class_getitem__, __missing__, __exit__, __await__,
+    # __del__, __aexit__, __getnewargs_ex__, __getnewargs__, __getstate__,
+    # __setstate__, __reduce__, __reduce_ex__
+    # since these either have no alternative method of being called or
+    # have a genuine use case for being called manually.
```

Comment:
As mentioned in the comment some of these (`__init__`, `__new__`, ...) have genuine use cases for being manually called.

On the other hand some are just excluded since they don't have an alternative method/function/operator for me to suggest.
e.g. you probably shouldn't be calling the `__del__` destructor directly and something like `__exit__` never needs to be manually called it just gets automatically executed when exiting a context manager.

If you'd like we could just have a message for these like `This is called automatically when exiting a context manager`, so we suggesting they remove it rather than replace it. I just excluded them for now as there could always be some weird unknown use case for these being called that we don't know about so playing it safe wrt false positives.

Easy either way on these ones 😄